Assets/Mirror/Runtime/NetworkWriterPool.cs

```diff
@@ -17,64 +17,31 @@ public void Dispose()
     /// <summary>
     /// Pool of NetworkWriters
     /// <para>Use this pool instead of <see cref="NetworkWriter">NetworkWriter</see> to reduce memory allocation</para>
-    /// <para>Use <see cref="Capacity">Capacity</see> to change size of pool</para>
     /// </summary>
     public static class NetworkWriterPool
     {
-        static readonly ILogger logger = LogFactory.GetLogger(typeof(NetworkWriterPool), LogType.Error);
-
-        /// <summary>
-        /// Size of the pool
-        /// <para>If pool is too small getting writers will causes memory allocation</para>
-        /// <para>Default value: 100 </para>
-        /// </summary>
-        public static int Capacity
-        {
-            get => pool.Length;
-            set
-            {
-                // resize the array
-                Array.Resize(ref pool, value);
-
-                // if capacity is smaller than before, then we need to adjust
-                // 'next' so it doesn't point to an index out of range
-                // -> if we set '0' then next = min(_, 0-1) => -1
-                // -> if we set '2' then next = min(_, 2-1) =>  1
-                next = Mathf.Min(next, pool.Length - 1);
-            }
-        }
-
-        /// <summary>
-        /// Mirror usually only uses up to 4 writes in nested usings,
-        /// 100 is a good margin for edge cases when users need a lot writers at
-        /// the same time.
-        ///
-        /// <para>keep in mind, most entries of the pool will be null in most cases</para>
-        /// </summary>
-        ///
-        /// Note: we use an Array instead of a Stack because it's significantly
-        ///       faster: https://github.com/vis2k/Mirror/issues/1614
-        static PooledNetworkWriter[] pool = new PooledNetworkWriter[100];
-
-        static int next = -1;
+        // reuse Pool<T>
+        // we still wrap it in NetworkWriterPool.Get/Recyle so we can reset the
+        // position before reusing.
+        // this is also more consistent with NetworkReaderPool where we need to
+        // assign the internal buffer before reusing.
+        //
+        // ushort.max to support all Transport.GetMaxMessageSizes!
+        // this way we don't depend & check on any transport size. it's const.
+        static readonly Pool<PooledNetworkWriter> pool
+            = new Pool<PooledNetworkWriter>(
+                () => new PooledNetworkWriter(ushort.MaxValue)
```

Comment:
Would be better to make this an option instead of constant value so it can be changed if needed by users